server/routes/uploadedImage.js

```diff
@@ -0,0 +1,179 @@
+const User = require("../models/user-schema");
+const express = require("express");
+const router = new express.Router();
+const auth = require("../auth");
+
+const mongoose = require('mongoose');
+const multer = require('multer');
+
+//Save to uploads folder and save image info
+const storage = multer.diskStorage({
+    destination: function(req, file, cb){
+        cb(null, './uploads/');
+    },
+    filename: function(req, file, cb){
+        cb(null, new Date().toISOString() + file.originalname);
+    }
+});
+
+const fileFilter = (req, file, cb) => {
+    //reject a file if not jpeg or png
+    if (file.mimetype === 'image/jpeg' || file.mimetype === 'image/png') {
+        cb(null, true);
+    } else {
+        cb(null, false);
+    }
+};
+
+// const upload = multer({dest: 'uploads/'});
+const upload = multer({
+    storage: storage,
+    limits: {
+        fileSize: 1024 * 1024 * 10 
+    },
+    fileFilter: fileFilter
+});
+ 
+const UploadedImage = require('../models/uploadedImage-schema');
+
+//Get all posts made for uploadedImage
+router.get('/', (req, res, next) => {
+    UploadedImage.find()
+    .select('name image _id uploadedImage')
+    .exec()
+    .then(docs => {
+        const response = {
+            count: docs.length,
+            images: docs.map(doc => {
+                return {
+                    name: doc.name,
+                    image: doc.image,
+                    uploadedImage: doc.uploadedImage,
+                    _id: doc._id,
+                    request: {
+                        type: 'GET',
+                        url: 'http://localhost:8000/api/uploadedImage/' + doc._id
+                    }
+                }
+            }) 
+        };
+        res.status(200).json(response);
+    })
+    .catch(err => {
+        console.log(err);
+        res.status(500).json({
+            error: err
+        });
+    });
+});
+
+//Post an image to the DB
+router.post('/', upload.single('uploadedImage'), (req, res, next) => {
```

Comment:
How does it know on who's account it should store a reference to the photo? For these requests, you should call the "auth" middleware as done in other routes. It will define req.user.id, and that can be used to find the account (aka the requesting user) to store the photo reference on.

Reply:
Yes, thats true. I created this as a template of sorts, to which I will then be adding auth as a middleware to identify for which user the uploads are for.